readthedocs/proxito/views/serve.py

```diff
@@ -673,3 +675,45 @@ def changefreqs_generator():
 
 class ServeSitemapXML(SettingsOverrideObject):
     _default_class = ServeSitemapXMLBase
+
+
+class ServeStaticFiles(CachedView, CacheTagsMixin, View):
+
+    """
+    Serve static files from the same domain the docs are being served from.
+
+    This is basically a proxy for ``STATIC_URL``.
+    """
+
+    project_cache_tag = "staticfile"
+    cache_request = True
+
+    @method_decorator(map_project_slug)
+    def get(self, request, filename, project):
+        try:
+            # This is needed for the _get_project
+            # method for the CacheTagsMixin class.
+            self.project = project
+            return FileResponse(staticfiles_storage.open(filename))
+        except FileNotFoundError:
+            # Don't use ``raise Http404`` so the other operations
+            # in ``.dispatch`` can be completed.
+            return HttpResponse(status=404)
+
+    def _get_cache_tags(self):
+        """
+        Add an additional *global* tag.
+
+        This is so we can purge all files from all projects
+        with one single call.
+        """
+        tags = super()._get_cache_tags()
+        tags.append(self.project_cache_tag)
```

Comment:
I have already reserved the project name `staticfile`, but we could also change these to be more specific, so they don't collide with project/version name as I suggested some time ago, like `+staticfile/project:+staticfile/project:+footer`

Reply:
I don't think collisions are a big deal, generally. But I don't think this is the place for changing that..